### PR TITLE
Add space between icon and text in a Button

### PR DESCRIPTION
**Pull request by doogthemushroom**
### Summary
Adds a small amount of space between an icon and the text in a Button. Was off putting to me that they were sitting right next to each other.

### Additional Details
I used 0.714em for the space to match the padding on the side of the buttons, but could see a smaller size being appropriate as well.

### CODE REVIEW

File: packages/terra-button/src/Button.scss [hunk 1]
```diff
@@ -117,4 +117,12 @@
...
     padding-top: var(--terra-button-padding-top-compact, 0);
   }
+
+  .text-with-icon {
+    padding-left: 0.714em;
+  }
+
+  .text-with-icon-reverse {
+    padding-right: 0.714em;
+  }
```
Comment (StephenEsser) [inline packages/terra-button/src/Button.scss]: I think these should be margins to push space between content

Comment (mhemesath) [reply packages/terra-button/src/Button.scss]: IMO we should have a component to manage this. Badges do basically the exact same thing.

Comment (bjankord) [reply packages/terra-button/src/Button.scss]: I'm working the OCS clinical/consumer teams to create a component for this. The consumer UX team started design work on this today.

File: packages/terra-button/src/Button.jsx [hunk 2]
```diff
@@ -80,7 +80,10 @@ const Button = ({
   ...customProps
   }) => {
   const attributes = Object.assign({}, customProps);
-  const buttonText = text ? <span className={styles.text}>{text}</span> : null;
+
+  const iconWithTextStyle = isReversed ? cx(['text-with-icon-reverse']) : cx(['text-with-icon']);
+  const textStyle = icon ? iconWithTextStyle : null;
+  const buttonText = text ? <span className={textStyle} >{text}</span> : null;
```
Comment (JakeLaCombe) [inline packages/terra-button/src/Button.jsx]: Extra space before >

File: packages/terra-button/src/Button.jsx [hunk 3]
```diff
@@ -80,7 +80,10 @@ const Button = ({
   ...customProps
   }) => {
   const attributes = Object.assign({}, customProps);
-  const buttonText = text ? <span className={styles.text}>{text}</span> : null;
+
+  const iconWithTextStyle = isReversed ? cx(['text-with-icon-reverse']) : cx(['text-with-icon']);
```
Comment (JakeLaCombe) [inline packages/terra-button/src/Button.jsx]: I think would be less confusing if we renamed it to iconTextStyleClass

Comment (StephenEsser) [reply packages/terra-button/src/Button.jsx]: Assuming the css for the Badge is robust we can just mimic the same css styling and keep this class as text and only make changes to Button.scss

https://github.com/cerner/terra-core/blob/master/packages/terra-badge/src/Badge.scss#L41-L47

**Comment by bjankord** [comment]
New OCS button designs had indicated spacing between icon and text. I'll follow up with the UX team and figure out what the values should be.

**Comment by bjankord** [comment]
Talked to UX and `0.714em` is the value they want for the spacing, nice work @doogthemushroom

**Comment by mjhenkes** [comment]
Actual footage of @doogthemushroom choosing the default button icon spacing.

**Comment by neilpfeiffer** [comment]
Tested: PR #947
 
**Reviewed:**
- Spacing between icon and button text looks good - ready to merge.

 
**Pass** |
-----------|
UX Review|

File: packages/terra-button/src/Button.jsx (hunk 2 above)
Comment (doogthemushroom) [reply packages/terra-button/src/Button.jsx]: Removed space

File: packages/terra-button/src/Button.scss (hunk 1 above)
Comment (doogthemushroom) [reply packages/terra-button/src/Button.scss]: Changed to margins.

Looking forward to the component.

File: packages/terra-button/src/Button.jsx (hunk 3 above)
Comment (doogthemushroom) [reply packages/terra-button/src/Button.jsx]: Ah that is some solid css. I've brought it in and removed the changes to the jsx.

**Pull request merged**